### ISSUE TITLE
Source-DB Schema in migration.ds.source.db.url has no effect, the schema is always 'dbo'

### ISSUE THREAD

**Issue by renatoh** [SAP-samples/commerce-migration-toolkit#12]
The migration always expects the source-DB schema to be dbo, regardless what is set on migration.ds.target.db.schema.
With that, the migration will not work, unless the schema of the source DB is converted to the default schema dbo.

I have noticed that the schema is not set to HikariConfig in org.sap.commercemigration.datasource.impl.DefaultMigrationDataSourceFactory#create

But even after adding this the line bellow, the problem persisted.
 config.setSchema(dataSourceConfiguration.getSchema());

**Comment by lnowakowski** [comment]
Please provide more details about source/target DB type. However source schema is mostly used for obtaining information from `INFORMATION_SCHEMA` (or other equivalent) to generate DDL adjustment script. Usually source schema for regular connection should be set directly in JDBC connection string

**Comment by renatoh** [comment]
ok, I forgot to mention, I did have the source schema also in the jdbc-connection url, it was like this:
migration.ds.source.db.url=jdbc:sqlserver://localhost:1433;databaseName=PQL;schema=PQL;responseBuffering=adaptive;trustServerCertificate=true;collation=Latin1_General_CI_AS;loginTimeout=10;

But when I inspect the schema form the connection to the source-DB, it is always dbo
![image](https://github.com/SAP-samples/commerce-migration-toolkit/assets/10344132/5b9d5b6a-92c1-41bb-bee0-e7c5e7f403e8)

**Comment by lnowakowski-sap** [comment]
From what I know, on MSSQL there is no option `schema` for connection string in JDBC driver, also do you have custom schema in your DB named PQL? As I mentioned - to fetch DB schema (structure) application is using value from: `context.getDataSourceRepository().getDataSourceConfiguration().getSchema()` not directly from connection details. See for example _AzureDataRepository#createAllTableNamesQuery()_

**Comment by renatoh** [comment]
yes, the schema is PQL, I am no able to set a configuration so that the I am able to connect to it, if I convert the schema to 'dbo' it works without any issues.

**Comment by lnowakowski-sap** [comment]
Maybe check this on DB user configuration, does your user has proper "Default schema" set?

**Comment by renatoh** [comment]
I think you are right, so I have to use a user with the source schema as default or as I did as a workaround converting the schema to dbo.
The connection check is, as it looks,  using the property for the source and target schema, e.g. migration.ds.source.db.schema, so the connection check is successful if I set the schemas accordingly, but then the migration is not using these properties and relies on the users default schema. That is very misleading.
